pkg/stanza/adapter/config.go

```diff
@@ -15,13 +15,43 @@
 package adapter // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/adapter"
 
 import (
+	"time"
+
 	"go.opentelemetry.io/collector/component"
 
 	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator"
 )
 
+// RetryConfig defines configuration for retrying batches in case of receiving a retryable error from a downstream
+// consumer. The current supported strategy is exponential backoff.
+type RetryConfig struct {
+	// Enabled indicates whether to not retry sending logs in case of receiving a retryable error from a downstream
+	// consumer. Default is false.
+	Enabled bool `mapstructure:"enabled"`
```

Comment:
Do you suggest that users use
```yaml
filelog:
  retry_on_failure:
```
instead of
```yaml
filelog:
  retry_on_failure:
    enabled: true
```

I know we use this pattern in a couple of places like otlp and hostmetrics receiver, but I don't think it should be applied here. I'd vote for having `retry_on_failure.enabled` as an explicit option. The first option seems a bit confusing to me as a user. Also, we follow the existing interface defined in https://github.com/open-telemetry/opentelemetry-collector/tree/main/exporter/exporterhelper#configuration

Reply:
`initial_interval`, `max_interval` and `max_elapsed_time` have reasonable defaults and should be considered as secondary configurations while `enable` is the most important one. I users had to configure all the secondary configurations, I don't think there is a chance they miss `enabled`

Reply:
This doesn't need to hold up the PR, but it seems like we should have a standard for when we need an `enabled` parameter.

Why do we use it in some places and not others? If it's better here, it seems we should be using it everywhere, unless there is a heuristic. I've added an item to the next SIG because I would like to understand this.